src/state/migrations.ts

```diff
@@ -0,0 +1,39 @@
+import { createMigrate, MigrationManifest, PersistedState, PersistMigrate } from 'redux-persist';
+import { MigrationConfig } from 'redux-persist/lib/createMigrate';
+
+function parseStorageItem(data: string | null) {
+  if (!data) return undefined;
+
+  try {
+    return JSON.parse(data);
+  } catch (_) {
+    return undefined;
+  }
+}
+
+export function customCreateMigrate(
+  migrations: MigrationManifest,
+  options: MigrationConfig
+): PersistMigrate {
+  const defaultMigrate = createMigrate(migrations, options);
+  return async (state: PersistedState, currentVersion: number) => {
+    if (state !== undefined) {
+      return defaultMigrate(state, currentVersion);
+    }
+
+    // Get all old localStorage items here
+    const oldTvChartConfig = parseStorageItem(localStorage.getItem('dydx.TradingViewChartConfig'));
+
+    // Remove (now) unused localStorage items
+    localStorage.removeItem('dydx.TradingViewChartConfig');
```

Comment:
IMO worth not removing old items until we finish QA (or even after things have been released for a day or two) in case there are issues.

it's not a big deal but if something is weird with the migration it might be nice to not lose locally stored transfer/trade history + notifications

Reply:
@yogurtandjam where in the code would you recommend removing `localStorage` items so that it still runs for everyone?

Reply:
hm actually, i think the way i wrote this, everyone new to redux-persist will start a version 0 and will run through the migrations, so might be fine to remove the localStorage components in a regular migration instead! i will include an example in this PR